drools-wb-screens/drools-wb-guided-rule-editor/drools-wb-guided-rule-editor-client/src/test/java/org/drools/workbench/screens/guided/rule/client/util/RefreshUtilTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.workbench.screens.guided.rule.client.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.gwt.i18n.client.DateTimeFormat;
+import com.google.gwtmockito.GwtMockitoTestRunner;
+import com.google.gwtmockito.WithClassesToStub;
+import org.drools.workbench.models.datamodel.rule.ActionFieldValue;
+import org.drools.workbench.models.datamodel.rule.BaseSingleFieldConstraint;
+import org.drools.workbench.models.datamodel.rule.FieldNatureType;
+import org.drools.workbench.models.datamodel.rule.SingleFieldConstraint;
+import org.drools.workbench.screens.guided.rule.client.editor.ActionValueEditor;
+import org.drools.workbench.screens.guided.rule.client.editor.ConstraintValueEditor;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+@WithClassesToStub(DateTimeFormat.class)
+@RunWith(GwtMockitoTestRunner.class)
+public class RefreshUtilTest {
```

Comment:
Is it worth doing a test where both drop-downs refresh?

These are used by "dependent" enums, e.g. Country/Region drop-downs where Region would refresh if Country changed. A more complex example would be Country/Region/Town. Changes to Country should refresh Region which then changes and should refresh Towns. A change to Region would not refresh Country. 

If you feel these tests are sufficient then ignore me (some tests are better than none!)

Reply:
@manstis I added some tests in this class, and also some formatting/imports cleanup, please could you re-review?